spikeextractors/extractors/neuroscopeextractors/neuroscopeextractors.py

```diff
@@ -410,20 +417,43 @@ def __init__(
                 for s_id in self._unit_ids:
                     self._spiketrains.append(res[(clu == s_id + 1).nonzero()])  # from 2,...,clu[0]-1
 
+        if spkfile_path is not None and Path(spkfile_path).is_file():
+            n_bits = int(xml_root.find('acquisitionSystem').find('nBits').text)
+            dtype = f"int{n_bits}"
+            n_samples = int(xml_root.find('neuroscope').find('spikes').find('nSamples').text)
+            wf = np.memmap(spkfile_path, dtype=dtype)
+            n_channels = int(wf.size / (n_spikes * n_samples))
+            wf = np.moveaxis(wf.reshape(n_spikes, n_samples, n_channels), 1, -1)
+
+            for unit_id in self.get_unit_ids():
+                if gain is not None:
+                    self.set_unit_property(unit_id=unit_id, property_name='gain', value=gain)
+                self.set_unit_spike_features(
+                    unit_id=unit_id,
+                    feature_name='waveforms',
+                    value=wf[clu == unit_id + 1 - int(keep_mua_units), :, :]
+                )
+
         if folder_path_passed:
             self._kwargs = dict(
                 resfile_path=None,
                 clufile_path=None,
                 folder_path=str(folder_path.absolute()),
-                keep_mua_units=keep_mua_units
+                keep_mua_units=keep_mua_units,
+                gain=gain
             )
         else:
             self._kwargs = dict(
                 resfile_path=str(resfile_path.absolute()),
                 clufile_path=str(clufile_path.absolute()),
                 folder_path=None,
-                keep_mua_units=keep_mua_units
+                keep_mua_units=keep_mua_units,
+                gain=gain
             )
+        if spkfile_path is not None:
+            self._kwargs.update(spkfile_path=str(spkfile_path.absolute()))
+        else:
+            self._kwargs.update(spkfile_path=spkfile_path)
```

Comment:
Do we want to set this if it's `None`?

Reply:
Yeah, dumped `kwargs` should always be set even if their default of `None`. Also present in code lines just above this, but this one is separated due to the fact that the `spkfile_path` argument itself does not interact with the `folder_path_passed` auto-detection scheme, since the very act of passing the `spkfile_path` acts like a boolean check that the user intends to load the waveform data in addition to the sorting data, which is a very optional choice.

Reply:
> dumped kwargs should always be set even if their default of None
why is that?

Reply:
Hmm... not actually sure, one would think that trying to load back a dumped extractor would just pass the kwargs dynamically so optionals would default if not present.

@alejoe91 Is there a more fundamental reason to have optional defaults set for `kwarg` dumping?